eth/query/block.go

```diff
@@ -81,12 +81,19 @@ func GetBlockByNumber(
 	blockInfo.LogsBloom = eth.EncBytes(bloomFilter)
 	for index, tx := range blockResult.Block.Data.Txs {
 		if full {
-			txResult, err := blockStore.GetTxResult(tx.Hash())
-			if err != nil {
-				return resp, errors.Wrapf(err, "cant find tx details, hash %X", tx.Hash())
+			var txResultBytes []byte
+			txResult, err := blockStore.GetBlockResults(&height)
```

Comment:
`height` does not change in the loop so `GetBlockResult` will return the same data, move this one out of `range blockResult.Block.Data.Txs` and change `txResult` to `blockResults`